js/src/common/states/ModalState.js

```diff
@@ -0,0 +1,94 @@
+import evented from '../utils/evented';
+
+import Modal from '../components/Modal';
+
+class ModalState {
+  constructor() {
+    this.clear();
+  }
+
+  /**
+   * Show a modal dialog.
+   *
+   * @param {Modal} component
+   * @public
+   */
+  show(modalClass, modalProps) {
+    if (!(modalClass.prototype instanceof Modal)) {
+      throw new Error('The ModalManager component can only show Modals');
+    }
+
+    clearTimeout(this.hideTimeout);
+
+    this.showing = true;
+    this.modalClass = modalClass;
+    this.modalProps = modalProps;
+
+    if (app.current) app.current.retain = true;
```

Comment:
This will have to go, but I think this will be taken care of at a later stage

Reply:
I'll remove it now, as it's no longer necessary as a result of the state change.